keypoints/include/pcl/keypoints/impl/sift_keypoint.hpp

```diff
@@ -171,6 +171,7 @@ pcl::SIFTKeypoint<PointInT, PointOutT>::detectKeypointsForOctave (
   findScaleSpaceExtrema (input, tree, diff_of_gauss, extrema_indices, extrema_scales);
 
   output.points.reserve (output.points.size () + extrema_indices.size ());
+  indices.indices.reserve (indices.indices.size () + extrema_indices.size ());
```

Comment:
This is not necessary, enough space is already reserved before the computation starts by the base class [here](https://github.com/PointCloudLibrary/pcl/blob/master/keypoints/include/pcl/keypoints/impl/keypoint.hpp#L117).